xml/System.Threading.Tasks/ValueTask`1.xml

```diff
@@ -661,9 +661,13 @@ As such, the default choice for any asynchronous method should be to return a <x
         <remarks>
           <format type="text/markdown"><![CDATA[  
 ## Remarks  
+ This method is intended for compiler use rather than for use in application code.
+
  If this <xref:System.Threading.Tasks.ValueTask%601> wraps a successful result, this property returns it directly.
 
- If it wraps a <xref:System.Threading.Tasks.Task%601>, the behavior of <xref:System.Threading.Tasks.ValueTask%601.Result%2A> is similar to the behavior of accessing <xref:System.Threading.Tasks.Task%601.Result%2A> on the wrapped task: if the task hasn't completed, accessing the property blocks the calling thread until it completes; if the task has completed successfully, the property returns the result; if the task has faulted or was cancelled, accessing the property throws an exception. The thrown exception is not wrapped in an <xref:System.AggregateException>, which is different from the behavior of <xref:System.Threading.Tasks.Task%601.Result%2A> in the same situation.
+ If this <xref:System.Threading.Tasks.ValueTask%601> wraps a <xref:System.Threading.Tasks.Task%601>, the behavior of <xref:System.Threading.Tasks.ValueTask%601.Result%2A> is similar to the behavior of accessing <xref:System.Threading.Tasks.Task%601.Result%2A> on the wrapped task: if the task hasn't completed, accessing the property blocks the calling thread until it completes; if the task has completed successfully, the property returns the result; if the task has faulted or was cancelled, accessing the property throws an exception. The thrown exception is not wrapped in an <xref:System.AggregateException>, which is different from the behavior of <xref:System.Threading.Tasks.Task%601.Result%2A> in the same situation.
+ 
+ If this <xref:System.Threading.Tasks.ValueTask%601> wraps a <xref:System.Threading.Tasks.Sources.IValueTaskSource%601>, the behavior of <xref:System.Threading.Tasks.ValueTask%601.Result%2A> depends on the behavior of the wrapped <xref:System.Threading.Tasks.Sources.IValueTaskSource%601>. Accessing this property may throw an exception if the <xref:System.Threading.Tasks.Sources.IValueTaskSource%601> has not completed.
```

Comment:
I'd like to be more strict than this and say that that the result should only be accessed when it's known that the ValueTask has already completed; the underlying IValueTaskSource may not expect a call in any other situation.  It also shouldn't be accessed more than once.

Reply:
I'm totally fine with that.

Also, I would encourage massively simplifying all of this. To me, what the `ValueTask<T>` is wrapping is an implementation-specific detail. Thinking of a first version of a library that wraps `Task<T>`, and then v2 wraps a `IValueTaskSource<T>`. In this scenario, `Result` in particular has a significant semantic change with no API differences to act as a warning.

So, a more ideal documentation for this property would be something simpler (and stricter) saying pretty much what you said, and completely ignoring "if this wraps..." situations.

Reply:
> So, a more ideal documentation for this property would be something simpler (and stricter) saying pretty much what you said, and completely ignoring "if this wraps..." situations.

I'm fine with that.